### PR TITLE
Extract client 'six' data types and use them for 'seven' data

### PR DESCRIPTION
**Pull request by Kaffeine**
## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by ChillerDragon** [comment]
I don't get it. Could you explain it for the lesser brained :brain:

**Comment by Kaffeine** [comment]
> I don't get it. Could you explain it for the lesser brained 🧠

Before this MR we had:
```cpp
// in game/generated/client_data.h:
struct CDataSound
{
	int m_Id;
	const char* m_pFilename;
};
struct CDataSoundset
{
	const char* m_pName;
	int m_NumSounds;
	CDataSound *m_aSounds;
	int m_Last;
};
// ...

// in game/generated/server_data.h:
struct CDataSound
{
	int m_Id;
	const char* m_pFilename;
};
struct CDataSoundset
{
	const char* m_pName;
	int m_NumSounds;
	CDataSound *m_aSounds;
	int m_Last;
};
// ...

// in game/generated/client_data7.h:
namespace client_data7 {
struct CDataSound
{
	int m_Id;
	const char* m_pFilename;
};
struct CDataSoundset
{
	const char* m_pName;
	int m_NumSounds;
	CDataSound *m_aSounds;
	int m_Last;
};
// ...

// in game/generated/server_data7.h:
namespace client_data7 {
struct CDataSound
{
	int m_Id;
	const char* m_pFilename;
};
struct CDataSoundset
{
	const char* m_pName;
	int m_NumSounds;
	CDataSound *m_aSounds;
	int m_Last;
};
// ...
```

Now we have:

```cpp
// in game/generated/data_types.h:
struct CDataSound
{
	int m_Id;
	const char* m_pFilename;
};
struct CDataSoundset
{
	const char* m_pName;
	int m_NumSounds;
	CDataSound *m_aSounds;
	int m_Last;
};
// ...

// in game/generated/client_data.h:
#include "data_types.h"
// ...

// in game/generated/server_data.h:
#include "data_types.h"
// ...

// in game/generated/client_data7.h:
#include "data_types.h"

namespace client_data7 {
// ...

// in game/generated/server_data7.h:
#include "data_types.h"

namespace client_data7 {
```

Client6 and client7 data now uses the same structs. There is no `client_data7::CDataSprite` anymore, just one shared `CDataSprite`.

**Comment by Kaffeine** [comment]
Of course it would be much better to also extract those classes from the two `content.py` to a shared file but I'm not a python dev and can't do that in reasonable time.

File: datasrc/content.py [hunk 1]
```diff
@@ -16,10 +16,11 @@ def __init__(self, name="", files=()):
 			self.sounds.Add(Sound(filename))
 
 class Image(Struct):
-	def __init__(self, name="", filename=""):
+	def __init__(self, name="", filename="", linear_mapping=0):
 		Struct.__init__(self, "CDataImage")
 		self.name = String(name)
 		self.filename = String(filename)
+		self.flag = Int(linear_mapping)
```
Comment (Robyt3) [inline datasrc/content.py:23]: Why are the names `flag` and `linear_mapping` different? Is that also on upstream?

Comment (Kaffeine) [reply datasrc/content.py:23]: That's what we have in https://github.com/ddnet/ddnet/blob/master/datasrc/seven/content.py#L23 and yes it is the same in https://github.com/teeworlds/teeworlds/blob/master/datasrc/content.py#L23.

**Review by Robyt3 — APPROVED** [review]

**Review by def- — APPROVED** [review]

**Pull request merged**